### PR TITLE
Clickhouse diagnostics

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `clickhouse-diagnostics` binary to the packages

### CODE REVIEW

**Comment by Felixoid** [comment]
@gingerwizard, unfortunately, I can't test the module inside docker. It tries to use docker as well.

```
name!@felix-carbon:/build/programs/diagnostics$ GOCACHE=/workdir/ make test
CLICKHOUSE_VERSION=latest go test -v -race `go list ./... | grep -v ./internal/platform/test`
?   	github.com/ClickHouse/ClickHouse/programs/diagnostics/cmd	[no test files]
?   	github.com/ClickHouse/ClickHouse/programs/diagnostics/cmd/clickhouse-diagnostics	[no test files]
=== RUN   TestNewParamMap
=== RUN   TestNewParamMap/test_param_map_correctly_converts_types
--- PASS: TestNewParamMap (0.00s)
    --- PASS: TestNewParamMap/test_param_map_correctly_converts_types (0.00s)
=== RUN   TestConvertParamsToConfig
=== RUN   TestConvertParamsToConfig/test_we_can_convert_a_param_map_back_to_a_config
--- PASS: TestConvertParamsToConfig (0.00s)
    --- PASS: TestConvertParamsToConfig/test_we_can_convert_a_param_map_back_to_a_config (0.00s)
=== RUN   TestAddParamMapToCmd
=== RUN   TestAddParamMapToCmd/test_we_can_add_hidden_params_to_a_command
--- PASS: TestAddParamMapToCmd (0.00s)
    --- PASS: TestAddParamMapToCmd/test_we_can_add_hidden_params_to_a_command (0.00s)
=== RUN   TestStringOptionsVar
=== RUN   TestStringOptionsVar/test_we_can_set
=== RUN   TestStringOptionsVar/test_set_invalid
--- PASS: TestStringOptionsVar (0.00s)
    --- PASS: TestStringOptionsVar/test_we_can_set (0.00s)
    --- PASS: TestStringOptionsVar/test_set_invalid (0.00s)
=== RUN   TestStringSliceOptionsVar
=== RUN   TestStringSliceOptionsVar/test_we_can_set
=== RUN   TestStringSliceOptionsVar/test_set_invalid
--- PASS: TestStringSliceOptionsVar (0.00s)
    --- PASS: TestStringSliceOptionsVar/test_we_can_set (0.00s)
    --- PASS: TestStringSliceOptionsVar/test_set_invalid (0.00s)
PASS
ok  	github.com/ClickHouse/ClickHouse/programs/diagnostics/cmd/params	0.035s
panic: failed to create container: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

goroutine 1 [running]:
github.com/ClickHouse/ClickHouse/programs/diagnostics/internal_test.TestMain(0xc000069168?)
	/build/programs/diagnostics/internal/runner_test.go:51 +0x607
main.main()
	_testmain.go:51 +0x318
FAIL	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal	0.023s
=== RUN   TestGetCollectorNames
=== RUN   TestGetCollectorNames/can_get_all_collector_names
=== RUN   TestGetCollectorNames/can_get_default_collector_names
--- PASS: TestGetCollectorNames (0.00s)
    --- PASS: TestGetCollectorNames/can_get_all_collector_names (0.00s)
    --- PASS: TestGetCollectorNames/can_get_default_collector_names (0.00s)
=== RUN   TestGetCollectorByName
=== RUN   TestGetCollectorByName/can_get_collector_by_name
=== RUN   TestGetCollectorByName/fails_on_non_existing_collector
--- PASS: TestGetCollectorByName (0.00s)
    --- PASS: TestGetCollectorByName/can_get_collector_by_name (0.00s)
    --- PASS: TestGetCollectorByName/fails_on_non_existing_collector (0.00s)
=== RUN   TestBuildConfigurationOptions
=== RUN   TestBuildConfigurationOptions/can_get_all_collector_configurations
--- PASS: TestBuildConfigurationOptions (0.00s)
    --- PASS: TestBuildConfigurationOptions/can_get_all_collector_configurations (0.00s)
PASS
ok  	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/collectors	0.040s
=== RUN   TestConfigConfiguration
=== RUN   TestConfigConfiguration/correct_configuration_is_returned_for_config_collector
--- PASS: TestConfigConfiguration (0.00s)
    --- PASS: TestConfigConfiguration/correct_configuration_is_returned_for_config_collector (0.00s)
=== RUN   TestConfigCollect
=== RUN   TestConfigCollect/test_default_file_collector_configuration
=== RUN   TestConfigCollect/test_configuration_when_specified
--- PASS: TestConfigCollect (0.00s)
    --- PASS: TestConfigCollect/test_default_file_collector_configuration (0.00s)
    --- PASS: TestConfigCollect/test_configuration_when_specified (0.00s)
=== RUN   TestDbLogsConfiguration
=== RUN   TestDbLogsConfiguration/correct_configuration_is_returned_for_summary_collector
--- PASS: TestDbLogsConfiguration (0.00s)
    --- PASS: TestDbLogsConfiguration/correct_configuration_is_returned_for_summary_collector (0.00s)
=== RUN   TestDbLogsCollect
=== RUN   TestDbLogsCollect/test_default_db_logs_collection
=== RUN   TestDbLogsCollect/test_db_logs_collection_with_limit
--- PASS: TestDbLogsCollect (0.00s)
    --- PASS: TestDbLogsCollect/test_default_db_logs_collection (0.00s)
    --- PASS: TestDbLogsCollect/test_db_logs_collection_with_limit (0.00s)
=== RUN   TestLogsConfiguration
=== RUN   TestLogsConfiguration/correct_configuration_is_returned_for_logs_collector
--- PASS: TestLogsConfiguration (0.00s)
    --- PASS: TestLogsConfiguration/correct_configuration_is_returned_for_logs_collector (0.00s)
=== RUN   TestLogsCollect
=== RUN   TestLogsCollect/test_default_logs_collection
=== RUN   TestLogsCollect/test_logs_collection_when_directory_is_specified
=== RUN   TestLogsCollect/test_logs_collection_of_archives
=== RUN   TestLogsCollect/test_when_directory_does_not_exist
--- PASS: TestLogsCollect (0.00s)
    --- PASS: TestLogsCollect/test_default_logs_collection (0.00s)
    --- PASS: TestLogsCollect/test_logs_collection_when_directory_is_specified (0.00s)
    --- PASS: TestLogsCollect/test_logs_collection_of_archives (0.00s)
    --- PASS: TestLogsCollect/test_when_directory_does_not_exist (0.00s)
=== RUN   TestSummaryConfiguration
=== RUN   TestSummaryConfiguration/correct_configuration_is_returned_for_summary_collector
--- PASS: TestSummaryConfiguration (0.00s)
    --- PASS: TestSummaryConfiguration/correct_configuration_is_returned_for_summary_collector (0.00s)
=== RUN   TestSummaryCollection
=== RUN   TestSummaryCollection/test_default_summary_collection
=== RUN   TestSummaryCollection/test_summary_collection_with_limit
--- PASS: TestSummaryCollection (0.02s)
    --- PASS: TestSummaryCollection/test_default_summary_collection (0.01s)
    --- PASS: TestSummaryCollection/test_summary_collection_with_limit (0.01s)
=== RUN   TestSystemConfiguration
=== RUN   TestSystemConfiguration/correct_configuration_is_returned_for_system_db_collector
--- PASS: TestSystemConfiguration (0.00s)
    --- PASS: TestSystemConfiguration/correct_configuration_is_returned_for_system_db_collector (0.00s)
=== RUN   TestSystemDbCollect
=== RUN   TestSystemDbCollect/test_default_system_db_collection
=== RUN   TestSystemDbCollect/test_when_we_pass_an_includes
=== RUN   TestSystemDbCollect/test_when_we_pass_an_excludes
=== RUN   TestSystemDbCollect/test_when_we_pass_an_invalid_includes
=== RUN   TestSystemDbCollect/test_when_we_use_a_table_with_excluded_fields
=== RUN   TestSystemDbCollect/test_with_a_low_row_limit
=== RUN   TestSystemDbCollect/test_with_a_negative_low_row_limit
=== RUN   TestSystemDbCollect/test_that_includes_overrides_excludes
=== RUN   TestSystemDbCollect/test_banned
=== RUN   TestSystemDbCollect/test_banned_unless_included
=== RUN   TestSystemDbCollect/tables_are_ordered_if_configured
--- PASS: TestSystemDbCollect (0.00s)
    --- PASS: TestSystemDbCollect/test_default_system_db_collection (0.00s)
    --- PASS: TestSystemDbCollect/test_when_we_pass_an_includes (0.00s)
    --- PASS: TestSystemDbCollect/test_when_we_pass_an_excludes (0.00s)
    --- PASS: TestSystemDbCollect/test_when_we_pass_an_invalid_includes (0.00s)
    --- PASS: TestSystemDbCollect/test_when_we_use_a_table_with_excluded_fields (0.00s)
    --- PASS: TestSystemDbCollect/test_with_a_low_row_limit (0.00s)
    --- PASS: TestSystemDbCollect/test_with_a_negative_low_row_limit (0.00s)
    --- PASS: TestSystemDbCollect/test_that_includes_overrides_excludes (0.00s)
    --- PASS: TestSystemDbCollect/test_banned (0.00s)
    --- PASS: TestSystemDbCollect/test_banned_unless_included (0.00s)
    --- PASS: TestSystemDbCollect/tables_are_ordered_if_configured (0.00s)
=== RUN   TestZookeeperConfiguration
=== RUN   TestZookeeperConfiguration/correct_configuration_is_returned_for_system_zookeeper_collector
--- PASS: TestZookeeperConfiguration (0.00s)
    --- PASS: TestZookeeperConfiguration/correct_configuration_is_returned_for_system_zookeeper_collector (0.00s)
=== RUN   TestZookeeperCollect
=== RUN   TestZookeeperCollect/test_default_zookeeper_collection
--- PASS: TestZookeeperCollect (0.00s)
    --- PASS: TestZookeeperCollect/test_default_zookeeper_collection (0.00s)
PASS
ok  	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/collectors/clickhouse	0.060s
=== RUN   TestCommandConfiguration
=== RUN   TestCommandConfiguration/correct_configuration_is_returned_for_file_collector
--- PASS: TestCommandConfiguration (0.00s)
    --- PASS: TestCommandConfiguration/correct_configuration_is_returned_for_file_collector (0.00s)
=== RUN   TestCommandCollect
=== RUN   TestCommandCollect/test_simple_command_with_args
[ls -l ../../../testdata total 0
drwxr-xr-x 6 1000 1000 62 Jun 30 16:25 configs
drwxr-xr-x 2 1000 1000 41 Jun 30 16:25 docker
drwxr-xr-x 3 1000 1000 17 Jun 28 16:00 logs
  ]
=== RUN   TestCommandCollect/test_empty_command
=== RUN   TestCommandCollect/test_invalid_command
--- PASS: TestCommandCollect (0.00s)
    --- PASS: TestCommandCollect/test_simple_command_with_args (0.00s)
    --- PASS: TestCommandCollect/test_empty_command (0.00s)
    --- PASS: TestCommandCollect/test_invalid_command (0.00s)
=== RUN   TestFileConfiguration
=== RUN   TestFileConfiguration/correct_configuration_is_returned_for_file_collector
--- PASS: TestFileConfiguration (0.00s)
    --- PASS: TestFileConfiguration/correct_configuration_is_returned_for_file_collector (0.00s)
=== RUN   TestFileCollect
=== RUN   TestFileCollect/test_filter_patterns_work
{"level":"debug","time":"2022-06-30T19:17:05Z","message":"Collecting file ../../../testdata/configs/include/xml/server-include.xml"}
{"level":"debug","time":"2022-06-30T19:17:05Z","message":"Collecting file ../../../testdata/configs/include/xml/user-include.xml"}
{"level":"debug","time":"2022-06-30T19:17:05Z","message":"Collecting file ../../../testdata/configs/xml/config.xml"}
{"level":"debug","time":"2022-06-30T19:17:05Z","message":"Collecting file ../../../testdata/configs/xml/users.xml"}
{"level":"debug","time":"2022-06-30T19:17:05Z","message":"Collecting file ../../../testdata/configs/xml/users.d/default-password.xml"}
{"level":"debug","time":"2022-06-30T19:17:05Z","message":"Collecting file ../../../testdata/configs/yandex_xml/config.xml"}
{"level":"debug","time":"2022-06-30T19:17:05Z","message":"Collecting file ../../../testdata/docker/admin.xml"}
{"level":"debug","time":"2022-06-30T19:17:05Z","message":"Collecting file ../../../testdata/docker/custom.xml"}
=== RUN   TestFileCollect/invalid_file_patterns_are_detected
=== RUN   TestFileCollect/check_empty_matches_are_reported
--- PASS: TestFileCollect (0.00s)
    --- PASS: TestFileCollect/test_filter_patterns_work (0.00s)
    --- PASS: TestFileCollect/invalid_file_patterns_are_detected (0.00s)
    --- PASS: TestFileCollect/check_empty_matches_are_reported (0.00s)
=== RUN   TestSystemConfiguration
=== RUN   TestSystemConfiguration/correct_configuration_is_returned_for_system_collector
--- PASS: TestSystemConfiguration (0.00s)
    --- PASS: TestSystemConfiguration/correct_configuration_is_returned_for_system_collector (0.00s)
=== RUN   TestSystemCollect
=== RUN   TestSystemCollect/test_default_system_collection
WARNING: failed to read disk partuuid of /dev/nvme0n1p1 : exit status 2
WARNING: failed to read disk partuuid of /dev/nvme0n1p2 : exit status 2
--- PASS: TestSystemCollect (0.04s)
    --- PASS: TestSystemCollect/test_default_system_collection (0.04s)
PASS
ok  	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/collectors/system	0.083s
=== RUN   TestGetOutputNames
=== RUN   TestGetOutputNames/can_get_all_output_names
--- PASS: TestGetOutputNames (0.00s)
    --- PASS: TestGetOutputNames/can_get_all_output_names (0.00s)
=== RUN   TestGetOutputByName
=== RUN   TestGetOutputByName/can_get_output_by_name
=== RUN   TestGetOutputByName/fails_on_non_existing_output
--- PASS: TestGetOutputByName (0.00s)
    --- PASS: TestGetOutputByName/can_get_output_by_name (0.00s)
    --- PASS: TestGetOutputByName/fails_on_non_existing_output (0.00s)
=== RUN   TestBuildConfigurationOptions
=== RUN   TestBuildConfigurationOptions/can_get_all_output_configurations
--- PASS: TestBuildConfigurationOptions (0.00s)
    --- PASS: TestBuildConfigurationOptions/can_get_all_output_configurations (0.00s)
PASS
ok  	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/outputs	0.022s
=== RUN   TestConfiguration
=== RUN   TestConfiguration/correct_configuration_is_returned
--- PASS: TestConfiguration (0.00s)
    --- PASS: TestConfiguration/correct_configuration_is_returned (0.00s)
=== RUN   TestWrite
=== RUN   TestWrite/test_we_can_write_simple_diagnostic_sets
{"level":"info","time":"2022-06-30T19:17:07Z","message":"creating bundle in /tmp/TestWritetest_we_can_write_simple_diagnostic_sets286536058/001/test/test"}
{"level":"info","time":"2022-06-30T19:17:07Z","message":"bundle created"}
=== RUN   TestWrite/test_invalid_parameter
=== RUN   TestWrite/test_compression
{"level":"info","time":"2022-06-30T19:17:07Z","message":"creating bundle in /tmp/TestWritetest_compression3319707675/001/test/test"}
{"level":"info","time":"2022-06-30T19:17:07Z","message":"bundle created"}
{"level":"info","time":"2022-06-30T19:17:07Z","message":"compressing bundle to /tmp/TestWritetest_compression3319707675/001/test/test.tar.gz"}
{"level":"info","time":"2022-06-30T19:17:07Z","message":"archive ready at: /tmp/TestWritetest_compression3319707675/001/test/test.tar.gz "}
=== RUN   TestWrite/test_support_for_directory_frames
{"level":"info","time":"2022-06-30T19:17:07Z","message":"creating bundle in /tmp/TestWritetest_support_for_directory_frames769288439/002/test/test"}
{"level":"info","time":"2022-06-30T19:17:07Z","message":"bundle created"}
=== RUN   TestWrite/test_support_for_config_frames
{"level":"info","time":"2022-06-30T19:17:07Z","message":"creating bundle in /tmp/TestWritetest_support_for_config_frames2955230202/002/test/test"}
{"level":"info","time":"2022-06-30T19:17:07Z","message":"bundle created"}
=== RUN   TestWrite/test_support_for_file_frames
{"level":"info","time":"2022-06-30T19:17:07Z","message":"creating bundle in /tmp/TestWritetest_support_for_file_frames360206908/002/test/test"}
{"level":"info","time":"2022-06-30T19:17:07Z","message":"bundle created"}
=== RUN   TestWrite/test_support_for_hierarchical_frames
{"level":"info","time":"2022-06-30T19:17:07Z","message":"creating bundle in /tmp/TestWritetest_support_for_hierarchical_frames554042755/001/test/test"}
{"level":"info","time":"2022-06-30T19:17:07Z","message":"bundle created"}
--- PASS: TestWrite (0.02s)
    --- PASS: TestWrite/test_we_can_write_simple_diagnostic_sets (0.00s)
    --- PASS: TestWrite/test_invalid_parameter (0.00s)
    --- PASS: TestWrite/test_compression (0.00s)
    --- PASS: TestWrite/test_support_for_directory_frames (0.00s)
    --- PASS: TestWrite/test_support_for_config_frames (0.01s)
    --- PASS: TestWrite/test_support_for_file_frames (0.00s)
    --- PASS: TestWrite/test_support_for_hierarchical_frames (0.00s)
PASS
ok  	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/outputs/file	0.039s
?   	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/outputs/terminal	[no test files]
panic: failed to create container: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

goroutine 1 [running]:
github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/platform_test.TestMain(0xc000180568?)
	/build/programs/diagnostics/internal/platform/manager_test.go:42 +0x6c7
main.main()
	_testmain.go:53 +0x318
FAIL	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/platform	0.030s
=== RUN   TestGetConfigParam
=== RUN   TestGetConfigParam/can_find_get_config_param_by_name
=== RUN   TestGetConfigParam/throws_error_on_missing_element
--- PASS: TestGetConfigParam (0.00s)
    --- PASS: TestGetConfigParam/can_find_get_config_param_by_name (0.00s)
    --- PASS: TestGetConfigParam/throws_error_on_missing_element (0.00s)
=== RUN   TestValidateConfig
=== RUN   TestValidateConfig/validate_adds_the_default_and_allows_override
=== RUN   TestValidateConfig/validate_errors_if_missing_param
=== RUN   TestValidateConfig/validate_errors_if_invalid_string_value
=== RUN   TestValidateConfig/allow_empty_string_value_if_specified
=== RUN   TestValidateConfig/validate_errors_if_invalid_string_options_value
=== RUN   TestValidateConfig/allow_empty_string_value_for_StringOptions_if_specified
--- PASS: TestValidateConfig (0.00s)
    --- PASS: TestValidateConfig/validate_adds_the_default_and_allows_override (0.00s)
    --- PASS: TestValidateConfig/validate_errors_if_missing_param (0.00s)
    --- PASS: TestValidateConfig/validate_errors_if_invalid_string_value (0.00s)
    --- PASS: TestValidateConfig/allow_empty_string_value_if_specified (0.00s)
    --- PASS: TestValidateConfig/validate_errors_if_invalid_string_options_value (0.00s)
    --- PASS: TestValidateConfig/allow_empty_string_value_for_StringOptions_if_specified (0.00s)
=== RUN   TestReadStringListValues
=== RUN   TestReadStringListValues/can_find_a_string_list_param
--- PASS: TestReadStringListValues (0.00s)
    --- PASS: TestReadStringListValues/can_find_a_string_list_param (0.00s)
=== RUN   TestReadStringValue
=== RUN   TestReadStringValue/can_find_a_string_param
--- PASS: TestReadStringValue (0.00s)
    --- PASS: TestReadStringValue/can_find_a_string_param (0.00s)
=== RUN   TestReadIntValue
=== RUN   TestReadIntValue/can_find_an_integer_param
--- PASS: TestReadIntValue (0.00s)
    --- PASS: TestReadIntValue/can_find_an_integer_param (0.00s)
=== RUN   TestReadBoolValue
=== RUN   TestReadBoolValue/can_find_a_boolean_param
--- PASS: TestReadBoolValue (0.00s)
    --- PASS: TestReadBoolValue/can_find_a_boolean_param (0.00s)
=== RUN   TestReadStringOptionsValue
=== RUN   TestReadStringOptionsValue/can_find_a_string_value_in_a_list_of_options
=== RUN   TestReadStringOptionsValue/errors_on_invalid_value
--- PASS: TestReadStringOptionsValue (0.00s)
    --- PASS: TestReadStringOptionsValue/can_find_a_string_value_in_a_list_of_options (0.00s)
    --- PASS: TestReadStringOptionsValue/errors_on_invalid_value (0.00s)
PASS
ok  	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/platform/config	0.022s
=== RUN   TestBundleError
=== RUN   TestBundleError/can_get_a_bundle_error
--- PASS: TestBundleError (0.00s)
    --- PASS: TestBundleError/can_get_a_bundle_error (0.00s)
=== RUN   TestString
=== RUN   TestString/can_order_by_asc
=== RUN   TestString/can_order_by_desc
--- PASS: TestString (0.00s)
    --- PASS: TestString/can_order_by_asc (0.00s)
    --- PASS: TestString/can_order_by_desc (0.00s)
=== RUN   TestNextDatabaseFrame
=== RUN   TestNextDatabaseFrame/can_iterate_sql_rows
=== RUN   TestNextDatabaseFrame/can_iterate_empty_sql_rows
--- PASS: TestNextDatabaseFrame (0.00s)
    --- PASS: TestNextDatabaseFrame/can_iterate_sql_rows (0.00s)
    --- PASS: TestNextDatabaseFrame/can_iterate_empty_sql_rows (0.00s)
=== RUN   TestNextFileDirectoryFrame
=== RUN   TestNextFileDirectoryFrame/can_iterate_file_frame
=== RUN   TestNextFileDirectoryFrame/can_iterate_file_frame_when_empty
--- PASS: TestNextFileDirectoryFrame (0.00s)
    --- PASS: TestNextFileDirectoryFrame/can_iterate_file_frame (0.00s)
    --- PASS: TestNextFileDirectoryFrame/can_iterate_file_frame_when_empty (0.00s)
=== RUN   TestNewConfigFileFrame
=== RUN   TestNewConfigFileFrame/can_iterate_config_file_frame
=== RUN   TestNewConfigFileFrame/can_iterate_file_frame_when_empty
--- PASS: TestNewConfigFileFrame (0.02s)
    --- PASS: TestNewConfigFileFrame/can_iterate_config_file_frame (0.01s)
    --- PASS: TestNewConfigFileFrame/can_iterate_file_frame_when_empty (0.00s)
=== RUN   TestConfigFileFrameCopy
=== RUN   TestConfigFileFrameCopy/can_copy_non-sensitive_xml_config_files
=== RUN   TestConfigFileFrameCopy/can_copy_sensitive_xml_config_files
=== RUN   TestConfigFileFrameCopy/can_copy_sensitive_yaml_config_files
--- PASS: TestConfigFileFrameCopy (0.14s)
    --- PASS: TestConfigFileFrameCopy/can_copy_non-sensitive_xml_config_files (0.02s)
    --- PASS: TestConfigFileFrameCopy/can_copy_sensitive_xml_config_files (0.07s)
    --- PASS: TestConfigFileFrameCopy/can_copy_sensitive_yaml_config_files (0.05s)
=== RUN   TestConfigFileFrameFindLogPaths
=== RUN   TestConfigFileFrameFindLogPaths/can_find_xml_log_paths
=== RUN   TestConfigFileFrameFindLogPaths/can_handle_empty_log_paths
=== RUN   TestConfigFileFrameFindLogPaths/can_find_yaml_log_paths
--- PASS: TestConfigFileFrameFindLogPaths (0.07s)
    --- PASS: TestConfigFileFrameFindLogPaths/can_find_xml_log_paths (0.03s)
    --- PASS: TestConfigFileFrameFindLogPaths/can_handle_empty_log_paths (0.00s)
    --- PASS: TestConfigFileFrameFindLogPaths/can_find_yaml_log_paths (0.04s)
=== RUN   TestYandexConfigFile
=== RUN   TestYandexConfigFile/can_find_xml_log_paths_with_yandex_root
--- PASS: TestYandexConfigFile (0.04s)
    --- PASS: TestYandexConfigFile/can_find_xml_log_paths_with_yandex_root (0.04s)
=== RUN   TestNextMemoryFrame
=== RUN   TestNextMemoryFrame/can_iterate_memory_frame
=== RUN   TestNextMemoryFrame/can_iterate_memory_frame_when_empty
=== RUN   TestNextMemoryFrame/can_iterate_memory_frame_when_empty#01
--- PASS: TestNextMemoryFrame (0.00s)
    --- PASS: TestNextMemoryFrame/can_iterate_memory_frame (0.00s)
    --- PASS: TestNextMemoryFrame/can_iterate_memory_frame_when_empty (0.00s)
    --- PASS: TestNextMemoryFrame/can_iterate_memory_frame_when_empty#01 (0.00s)
PASS
ok  	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/platform/data	0.291s
panic: failed to create container: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

goroutine 1 [running]:
github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/platform/database_test.TestMain(0xc000202168?)
	/build/programs/diagnostics/internal/platform/database/native_test.go:44 +0x607
main.main()
	_testmain.go:57 +0x318
FAIL	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/platform/database	0.032s
panic: failed to create container: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

goroutine 1 [running]:
github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/platform/utils_test.TestMain(0xc000057168?)
	/build/programs/diagnostics/internal/platform/utils/process_test.go:40 +0x5e7
main.main()
	_testmain.go:69 +0x318
FAIL	github.com/ClickHouse/ClickHouse/programs/diagnostics/internal/platform/utils	0.025s
FAIL
make: *** [Makefile:41: test] Error 1
```

Can you provide a tag to disable dockerized tests?

**Comment by Felixoid** [comment]
Fixed typos issues in style check, see https://s3.amazonaws.com/clickhouse-test-reports/38647/2edb6ec6b5daa3d1b960122549b9c5f86262f216/style_check__actions_.html

**Comment by gingerwizard** [comment]
@Felixoid i thought docker in docker was fine provided you mount docker.sock

**Comment by Felixoid** [comment]
I launch the tests in the same container we use for building. It's the least effort for now.

Making a dedicated job for a proper go test is a magnitude more job comparing to what is done here

**Comment by Felixoid** [comment]
After the tests are finished, it's ready for review. The built binary should be in the report as well https://github.com/ClickHouse/ClickHouse/runs/7149504573?check_suite_focus=true#step:7:13577

File: docker/packager/binary/build.sh [hunk 1]
```diff
@@ -29,8 +29,21 @@ env
 
 if [ -n "$MAKE_DEB" ]; then
   rm -rf /build/packages/root
+  if [ -z "$SANITIZER" ]; then
+    # We need to check if clickhouse-diagnostics is fine and build it
+    (
+      cd /build/programs/diagnostics
+      make test-no-docker
+      GOARCH="${DEB_ARCH}" CGO_ENABLED=0 make VERSION="$CLICKHOUSE_VERSION_STRING" build
```
Comment (gingerwizard) [inline docker/packager/binary/build.sh:37]: Can we pass in ldflags here with the version? It ensures the binary can compile in the version and will report the correct version with the user does `clickhouse-diagnostics version`. see https://github.com/ClickHouse/clickhouse-diagnostics/blob/main/Makefile#L8.
I think we can align with the clickhouse version?

Comment (gingerwizard) [reply docker/packager/binary/build.sh:37]: ah nvm i see you do below.

**Review by gingerwizard — APPROVED** [review]
LGTM

**Comment by Felixoid** [comment]
wait a second =)

**Comment by Felixoid** [comment]
`Jul 04 09:56:07 go build -ldflags "-X github.com/ClickHouse/ClickHouse/programs/diagnostics/cmd.Version=22.7.1.1 -X github.com/ClickHouse/ClickHouse/programs/diagnostics/cmd.Commit=7982c66a" -o clickhouse-diagnostics ./cmd/clickhouse-diagnostics`

https://github.com/ClickHouse/ClickHouse/runs/7178169708?check_suite_focus=true#step:7:893

Looks good

**Pull request merged**